mozregression/test_runner.py

```diff
@@ -13,37 +13,41 @@
 import os
 import datetime
 
-from mozregression.launchers import create_launcher
+from mozregression.launchers import create_launcher as mozlauncher
 from mozregression.errors import TestCommandError, LauncherError
+from abc import ABCMeta, abstractmethod
 
 LOG = get_proxy_logger("Test Runner")
 
 
-class TestRunner(object):
+def create_launcher(build_info):
+    """
+    Create and returns a :class:`mozregression.launchers.Launcher`.
+    """
+    if build_info.build_type == 'nightly':
+        if isinstance(build_info.build_date, datetime.datetime):
+            desc = ("for buildid %s"
+                    % build_info.build_date.strftime("%Y%m%d%H%M%S"))
+        else:
+            desc = "for %s" % build_info.build_date
+    else:
+        desc = ("built on %s, revision %s"
+                % (build_info.build_date,
+                   build_info.short_changeset))
+    LOG.info("Running %s build %s" % (build_info.repo_name, desc))
+
+    return mozlauncher(build_info)
```

Comment:
hmm, I had thought about this too. But it means we may need to import whole launcher like "from mozregression import launcher", however we only use creat_launcher for it.

We can merge this for now and to see if we need do some follow up work for it anyway :)